contracts/test/integration/vip/vip14.sol

```diff
@@ -0,0 +1,453 @@
+//SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity =0.8.13;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {TimelockController} from "@openzeppelin/contracts/governance/TimelockController.sol";
+
+import {Vm} from "./../../unit/utils/Vm.sol";
+import {Core} from "../../../core/Core.sol";
+import {IVIP} from "./IVIP.sol";
+import {DSTest} from "./../../unit/utils/DSTest.sol";
+import {PCVDeposit} from "../../../pcv/PCVDeposit.sol";
+import {PCVGuardian} from "../../../pcv/PCVGuardian.sol";
+import {IPCVDeposit} from "../../../pcv/IPCVDeposit.sol";
+import {PriceBoundPSM} from "../../../peg/PriceBoundPSM.sol";
+import {ERC20Allocator} from "../../../pcv/utils/ERC20Allocator.sol";
+import {MaplePCVDeposit} from "../../../pcv/maple/MaplePCVDeposit.sol";
+import {VoltSystemOracle} from "../../../oracle/VoltSystemOracle.sol";
+import {MainnetAddresses} from "../fixtures/MainnetAddresses.sol";
+import {ArbitrumAddresses} from "../fixtures/ArbitrumAddresses.sol";
+import {OraclePassThrough} from "../../../oracle/OraclePassThrough.sol";
+import {CompoundPCVRouter} from "../../../pcv/compound/CompoundPCVRouter.sol";
+import {ITimelockSimulation} from "../utils/ITimelockSimulation.sol";
+import {MorphoCompoundPCVDeposit} from "../../../pcv/morpho/MorphoCompoundPCVDeposit.sol";
+
+/// Deployment Steps
+/// 1. deploy morpho dai deposit
+/// 2. deploy morpho usdc deposit
+/// 3. deploy compound pcv router pointed to morpho dai and usdc deposits
+/// 4. deploy volt system oracle
+/// 5. deploy maple usdc deposit
+
+/// Governance Steps
+/// 1. grant new PCV router PCV Controller role
+/// 2. revoke PCV Controller role from old PCV Router
+
+/// 3. disconnect old dai compound deposit from allocator
+/// 4. disconnect old usdc compound deposit from allocator
+
+/// 5. connect new dai morpho deposit to allocator
+/// 6. connect new usdc morpho deposit to allocator
+
+/// 7. add deposits as safe addresses
+
+/// 8. connect new oracle to oracle pass through with updated rate
+
```

Comment:
9. grant timelock pcv controller role
9.1 call deposit on maple deposit
9.2 revoke pcv controller role from timelock

Reply:
no need to revoke